packages/muelu/src/Misc/MueLu_SegregatedAFactory_decl.hpp

```diff
@@ -47,26 +47,54 @@
+    @brief  Factory for building a new "segregated" A operator. Here, "segregated" means that the user
+            provides some map(s) (containing a subset of GIDs of the input matrix A) and the factory
+            drops entries depending on the dropping scheme.
+
+            ## Idea ##
+
+            The idea is to use the output matrix A as input for the aggregation factory to have control over
+            the aggregates and make sure that aggregates do not cross certain areas.
+
+            ## Remarks ##
+
+            This factory supports multiple dropping Schemes based on different inputs. They are:
+
+            - blockmap: Based on the user provided "blockmap", the off-diagonal entries (a,b) and (b,a) in A are dropped.
+            "a" denotes a GID entry in the provided map and "b" denotes a GID that is not contained in the provided map.
+            In this use case the Factory expects a dropMap1 (==blockmap).
+            The blockmap scheme also doesn't support the "Call ReduceAll on dropMap1/2" options.
+
+            - map-pair: Based on a "map-pair", the user provides two maps "dropMap1" and "dropMap2",
+            which specify global row/column pairs in the operator A to be dropped.
+            The Factory drops any possible combination of the dropMaps 1 and 2. To ensure that entry A(a,b) is
+            dropped, as well as entry A(b,a), there is an option to create redundant dropMaps on all Procs.
+            This ensures that entries aren't overlooked due to the local rowmaps of the operator A.
+
+            Note: we have to drop the entries (i.e. not just set them to zero) as the CoalesceDropFactory
+                 does not distinguish between matrix entries which are zero and nonzero.
 
-         The idea is to use the output matrix A as input for the aggregation factory to have control over
-         the aggregates and make sure that aggregates do not cross certain areas.
+    ## Input/output of this factory ##
 
-         Note: we have to drop the entries (i.e. not just set them to zero) as the CoalesceDropFactory
-               does not distinguish between matrix entries which are zero and nonzero.
+    ### User parameters of SegregatedAFactory ###
+    Parameter | type | default | master.xml | validated | requested | description
+    ----------|------|---------|:----------:|:---------:|:---------:|------------
+    A             | Factory | null  |   | * | * | Generating factory of the matrix A
+    droppingScheme| string  | vague |   | * | * | Strategy to drop entries from matrix A based on the input of some map(s) [blockmap, map-pair]
+    dropMap1      | Factory | null  |   | * | * | Generating factory for dropMap1
+    dropMap2      | Factory | null  |   | * | * | Generating factory for dropMap2
+    Call ReduceAll on dropMap1 | bool |   | * |  | Boolean for calling reduceAll on dropMap1
+    Call ReduceAll on dropMap2 | bool |   | * |  | Boolean for calling reduceAll on dropMap2
```

Comment:
What does `reduceAll` do?

Reply:
I tried to explain this in the documentation:

> - map-pair: Based on a "map-pair", the user provides two maps "dropMap1" and "dropMap2",
            which specify global row/column pairs in the operator A to be dropped.
            The Factory drops any possible combination of the dropMaps 1 and 2. To ensure that entry A(a,b) is
            dropped, as well as entry A(b,a), there is an option to create redundant dropMaps on all Procs.
            This ensures that entries aren't overlooked due to the local rowmaps of the operator A.

In our application there are cases, in which entries were copied, although they should have been. The only way I was able to circumvent this was by creating redundant dropMaps.

Reply:
Here is my interpretation. We want to drop entries `A(a,b)` and `A(b,a)` for GIDs `a` and `b` in the two maps. The maps are submaps of the row map. But we have some off-rank entries in the column map. If we use the maps as they are we are missing these off-rank entries. Is that what's going on here? 
If so, doing an allReduce to get the entire map on one each rank seems overkill. We really only want to do an import of the dropping information into the column map. One way to achieve this is to create an `Xpetra::Vector<int>`, set dropped entries to 1 and un-dropped to 0 and import that into the column map. If this isn't clear feel free to ping me and we'll set up a quick call.

Reply:
Thank you for the input! I finally had time to take a look at the implementation and have come up with what a utility which should basically do this. I have added it to MueLu Utilities and added a unittest.